support/util/containers.go

```diff
@@ -139,3 +140,49 @@ func WithOptions(opts *AvailabilityProberOpts) AvailabilityProberOpt {
 		o.WaitForClusterRolebinding = opts.WaitForClusterRolebinding
 	}
 }
+
+// enforceRestrictedSecurityContext enforces restricted security context settings on a single container.
+// Per Kubernetes restricted pod security standards, only NET_BIND_SERVICE is allowed.
+// See: https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted
+func enforceRestrictedSecurityContext(container *corev1.Container) error {
+	if container.SecurityContext == nil {
+		container.SecurityContext = &corev1.SecurityContext{}
+	}
+	container.SecurityContext.AllowPrivilegeEscalation = ptr.To(false)
+	container.SecurityContext.RunAsNonRoot = ptr.To(true)
+
+	var existingAdd []corev1.Capability
+	if container.SecurityContext.Capabilities != nil {
+		existingAdd = container.SecurityContext.Capabilities.Add
+		// Validate capabilities against restricted pod security standards
+		for _, cap := range existingAdd {
+			if cap != "NET_BIND_SERVICE" {
+				return fmt.Errorf("container %q: capability %q is not allowed by restricted pod security standards (only NET_BIND_SERVICE is permitted)", container.Name, cap)
```

Comment:
so this will fail in runtime if any container defines this cap in it's manifest, right?
Ideally this should be caught beforehand in unit test. 
Please add a test case for GCP platform in `TestControlPlaneComponents` similar to IBM here: 
https://github.com/openshift/hypershift/blob/95835694eb99c9afc6207738d64a30db9f37e83b/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller_test.go#L1799-L1803

Reply:
Yes that's the idea. If we don't fail, GKE will reject the pod/deployment anyways later.

I'll add this test.